packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako

```diff
@@ -187,3 +228,11 @@ class ${class_name}:
 
         for response in self._get_stub().Measure(request):
             yield self._deserialize_response(response)
+
+    def register_pin_map(self, pin_map_path: str) -> None:
```

Comment:
BTW, an optional `pin_map_context` parameter would allow users to use a single client object to make multiple concurrent calls to the same service with different sites. However, we have already agreed to put cancellation state in the client object, so it's not necessary to make the client object entirely stateless.

Reply:
> users to use a single client object to make multiple concurrent calls to the same service with different sites

As we discussed in [HLD](https://dev.azure.com/ni/DevCentral/_git/ASW/pullRequest/751696#1721325101), the current implementation does not support multiple concurrent calls from a single client object. Instead, calls must be made sequentially, one after the other. 

So, I've considered option 1 of adding a property for `pin_map_context` with getter and setter. This way users can update the `pin_map_context` for each measure call. For example, like below

```
    client = NIDCPowerSourceDCVoltageClient()
    client.pin_map_context = PinMapContext(pin_map_id="NIDCPowerSourceDCVoltage.pinmap", sites=[0])
    client.measure()
    client.pin_map_context = PinMapContext(pin_map_id="NIDCPowerSourceDCVoltageMultiSite.pinmap", sites=[1])
    client.measure()
```

Reply:
ok, and the cancellation PR enforces this